src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -137,43 +131,35 @@ public async Task EndInvokeJSFromDotNet(long asyncCall, bool succeded, string ar
             }
         }
 
-        public async Task DispatchEvent(string eventDescriptorJson, string eventArgs)
+        public async Task DispatchEvent(string eventDescriptorJson, string eventArgsJson)
         {
-            RendererRegistryEventDispatcher.BrowserEventDescriptor eventDescriptor = null;
+            WebEventData webEventData;
             try
             {
                 AssertInitialized();
-                eventDescriptor = ParseEventDescriptor(eventDescriptorJson);
-                if (eventDescriptor == null)
-                {
-                    return;
-                }
-
-                await Renderer.Dispatcher.InvokeAsync(() =>
-                {
-                    SetCurrentCircuitHost(this);
-                    return RendererRegistryEventDispatcher.DispatchEvent(eventDescriptor, eventArgs);
-                });
+                webEventData = new WebEventData(eventDescriptorJson, eventArgsJson);
             }
             catch (Exception ex)
             {
-                Log.DispatchEventFailedToDispatchEvent(_logger, eventDescriptor != null ? eventDescriptor.EventHandlerId.ToString() : null, ex);
-                UnhandledException?.Invoke(this, new UnhandledExceptionEventArgs(ex, isTerminating: false));
+                Log.DispatchEventFailedToParseEventData(_logger, ex);
+                return;
```

Comment:
This should be still be considered a terminal error `UnhandledException`. I'll be taking a second look at this in my PR - but is there a reason why you removed it?

Reply:
Ignore that.